src/CoreGraphics/CGPDFOperatorTable.cs

```diff
@@ -71,20 +71,35 @@ protected virtual void Dispose (bool disposing)
 
 		public IntPtr Handle { get; private set; }
 
+		// We need this P/Invoke for legacy AOT scenarios (since we have public API taking a 'Action<IntPtr, IntPtr>', and with this particular native function we can't wrap the delegate)
+		// Unfortunately CoreCLR doesn't support generic Action delegates in P/Invokes: https://github.com/dotnet/runtime/issues/32963
 		[DllImport (Constants.CoreGraphicsLibrary)]
 		extern static void CGPDFOperatorTableSetCallback (/* CGPDFOperatorTableRef */ IntPtr table, /* const char */ string name, /* CGPDFOperatorCallback */ Action<IntPtr,IntPtr> callback);
 
+#if NET
+		// This signature requires C# 9 (so .NET only).
+		// The good part about this signature is that it enforces at compile time that 'callback' is callable from native code in a FullAOT scenario.
+		// The bad part is that it's unsafe code (and callers must be in unsafe mode as well).
+		[DllImport (Constants.CoreGraphicsLibrary)]
+		unsafe extern static void CGPDFOperatorTableSetCallback (/* CGPDFOperatorTableRef */ IntPtr table, /* const char */ string name, /* CGPDFOperatorCallback */ delegate* unmanaged<IntPtr, IntPtr, void> callback);
+#endif
+
 #if MONOMAC
+		// This signature can work everywhere, but we can't enforce at compile time that 'callback' is a delegate to a static function (which is required for FullAOT scenarios),
+		// so limit it to non-FullAOT platforms (macOS)
+		[DllImport (Constants.CoreGraphicsLibrary)]
+		extern static void CGPDFOperatorTableSetCallback (/* CGPDFOperatorTableRef */ IntPtr table, /* const char */ string name, /* CGPDFOperatorCallback */ CGPDFOperatorCallback callback);
+
 		// this won't work with AOT since the callback must be decorated with [MonoPInvokeCallback]
 		public void SetCallback (string name, Action<CGPDFScanner,object> callback)
 		{
 			if (name == null)
 				throw new ArgumentNullException ("name");
 
 			if (callback == null)
-				CGPDFOperatorTableSetCallback (Handle, name, null);
+				CGPDFOperatorTableSetCallback (Handle, name, (CGPDFOperatorCallback) null);
```

Comment:
Woah I don't recall seeing this before haha. A type cast for null! What does this do?

Reply:
@tj-devel709 imaging you are a compiler and you see:
```
public void CGPDFOperatorTableSetCallback (IntPtr handel, String name, CGPDFOperatorCallback callback) {...}
public void CGPDFOperatorTableSetCallback (IntPtr handel, String name, OtherObject obj) {..}
```

Later in the code I have:
```csharp
CGPDFOperatorTableSetCallback (Handle, name, null);
```

Which of the above overloads do you call??

And if you see:
```csharp
CGPDFOperatorTableSetCallback (Handle, name, (CGPDFOperatorCallback) null);
```

AFAIK you can also do:
```csharp
CGPDFOperatorTableSetCallback (Handle, name, default(CGPDFOperatorCallback));
```

which helps the compiler too without a cast.

Reply:
Ah thank you for explaining!